### PR TITLE
Bump GH actions (and probably dependencies)

### PR DESCRIPTION
**Pull request by hollabaq86**
Bumps the Github actions file to newest versions of Node that are still actively supported (Node 14 will go EOL in two months, not worth including it IMO).
Makes sure GH actions run on pull requests and not just commits being pushed.
Ran `npm audit` and updated packages

### CODE REVIEW

**Review by crookedneighbor — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
         with:
-          node-version: '14.x'
-      - run: npm install
-      - run: npm run test:unit
-      - run: npm start & sleep 3 && npm run test:integration
+          node-version: {{ matrix.node-version }}
+
+      - run: npm ci
+
+      - name: Run tests
```
Comment (crookedneighbor) [inline .github/workflows/ci.yml:28]:
```suggestion
```

I think there can only be one `name` in this block.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
         with:
-          node-version: '14.x'
-      - run: npm install
-      - run: npm run test:unit
-      - run: npm start & sleep 3 && npm run test:integration
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm ci
+
+      - run: npm run test
```
Comment (crookedneighbor) [inline .github/workflows/ci.yml:28]:
```suggestion
      - run: npm ci
      - run: npm run test:unit
      - run: npm start & sleep 3 && npm run test:integration
```

Need to do it this way so the test server starts up.

Comment (hollabaq86) [reply .github/workflows/ci.yml:28]: ahhhhhhhhhhhh

**Review by crookedneighbor — APPROVED** [review]

**Review by SancSalix — APPROVED** [review]
🌟

**Pull request merged**